serving/samples/helloworld-haskell/README.md

```diff
@@ -0,0 +1,198 @@
+apps are generally more useful if you build them step-by-step. The
+following instructions recreate the source files from this folder.
+
+1. Create a new file named `stack.yaml` and paste the following code:
+
+    ```yaml
+    flags: {}
+    packages:
+    - .
+    extra-deps: []
+    resolver: lts-10.7
+    ```
+1. Create a new file named `package.yaml` and paste the following code
+
+	```yaml
+    name:                helloworld-haskell
+    version:             0.1.0.0
+    dependencies:
+    - base >= 4.7 && < 5
+    - scotty
+    - text
+
+    executables:
+      helloworld-haskell-exe:
+        main:                Main.hs
+        source-dirs:         app
+        ghc-options:
+        - -threaded
+        - -rtsopts
+        - -with-rtsopts=-N
+	```
+
+1. Create a `app` folder, then create a new file named `Main.hs` in that folder
+   and paste the following code. This code creates a basic web server which
+   listens on port 8080:
+
+    ```haskell
+	{-# LANGUAGE OverloadedStrings #-}
+
+	import           Data.Maybe
+	import           Data.Monoid        ((<>))
+	import           Data.Text.Lazy     (Text)
+	import           Data.Text.Lazy
+	import           System.Environment (lookupEnv)
+	import           Web.Scotty         (ActionM, ScottyM, scotty)
+	import           Web.Scotty.Trans
+
+	main :: IO ()
+	main = do
+      t <- fromMaybe "NOT SPECIFIED" <$> lookupEnv "TARGET"
+      scotty 8080 (route t)
+
+	route :: String -> ScottyM()
+	route t = get "/" $ hello t
+
+	hello :: String -> ActionM()
+	hello t = text $ pack ("Hello world: " ++ t)
+    ```
+
+1. In your project directory, create a file named `Dockerfile` and copy the code
+   block below into it.
+
+    ```docker
+	# Use the existing Haskell image as our base
+	FROM haskell:8.2.2 as builder
+
+	# Checkout our code onto the Docker container
+	WORKDIR /app
+	ADD . /app
+
+	# Build and test our code, then install the “helloworld-haskell-exe” executable
+	RUN stack setup
+	RUN stack build --copy-bins
+
+	# Copy the "helloworld-haskell-exe" executable to the image using docker multi stage build
+	FROM fpco/haskell-scratch:integer-gmp
+	WORKDIR /root/
+	COPY --from=builder /root/.local/bin/helloworld-haskell-exe .
+
+	# Expose a port to run our application
+	EXPOSE 8080
+
+	# Run the server command
+	CMD ["./helloworld-haskell-exe"]
+    ```
+
+1. Create a new file, `service.yaml` and copy the following service definition
+   into the file. Make sure to replace `{username}` with your Docker Hub username.
+
+    ```yaml
+    apiVersion: serving.knative.dev/v1alpha1
+    kind: Service
+    metadata:
+    name: helloworld-haskell
+    namespace: default
+    spec:
+    runLatest:
+        configuration:
+        revisionTemplate:
+            spec:
+            container:
+                image: docker.io/{username}/helloworld-haskell
+                env:
+                - name: TARGET
+                value: "Haskell Sample v1"
+    ```
```

Comment:
The indentation here is off. Copying and pasting from here, I couldn't apply the service.yaml without getting an error. When I went and copied the raw version of the file, the service was successfully created.

I would re-copy from your service.yaml file to get the indentation right.

Reply:
I will work on this today. Thanks for the inputs.